language/move-vm/runtime/src/loader.rs

```diff
@@ -412,22 +412,22 @@ impl ModuleCache {
 //
 
 // A Loader is responsible to load scripts and modules and holds the cache of all loaded
-// entities. Each cache is protected by a `Mutex`. Operation in the Loader must be thread safe
+// entities. Each cache is protected by a `RwLock`. Operation in the Loader must be thread safe
 // (operating on values on the stack) and when cache needs updating the mutex must be taken.
 // The `pub(crate)` API is what a Loader offers to the runtime.
 pub(crate) struct Loader {
-    scripts: Mutex<ScriptCache>,
-    module_cache: Mutex<ModuleCache>,
-    type_cache: Mutex<TypeCache>,
+    scripts: RwLock<ScriptCache>,
```

Comment:
When is a single `Loader` shared by multiple threads?

Reply:
Signature verification IIRC

Reply:
Why does signature verification need to look at scripts/modules/types? Shouldn't it only need to inspect a public key, a signature, and a message?

Reply:
@runtian-zhou still wondering about this if you happen to know...

Reply:
Ah sry. The loader would be used in a multi threaded context in two scenarios: 
1. The VMValidator will spawn multiple threads to the same VM to validate a transaction
2. The parallel executor might spawn threads to execute a transaction. This is more of a future proof use case and currently we discover that having each thread to own a unique instance of VM is simply faster.